include/ck/utility/data_type.hpp

```diff
@@ -996,6 +1411,18 @@ inline __host__ __device__ constexpr bhalf_t type_convert<bhalf_t, float>(float
     return uint16_t(u.int32 >> 16);
 }
 
+template <>
+inline __host__ __device__ constexpr int8_t type_convert<int8_t, int4_t>(int4_t x)
+{
+    return int8_t(x);
+}
+
+template <>
+inline __host__ __device__ constexpr int4_t type_convert<int4_t, int8_t>(int8_t x)
+{
+    return int4_t(x & 0x7);
```

Comment:
How about the sign bit of `int4_t` ?

Reply:
Ultimately we will probably have some kind of quantization between those two types. @zjing14 has more details.